### PR TITLE
qa/cephfs: correct the parameters' order 

### PR DESCRIPTION
**Pull request by lxbsz**
The parameters' order is incorrect and missing the client_config.

Fixes: https://tracker.ceph.com/issues/53216
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [ ] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by batrick — DISMISSED** [review]
Please add

`Fixes: 242585656c6bc282f5adbd073e83bafa86b5c0d2`

to the relevant commit. That's used by backporters to find regression fixes.

**Comment by lxbsz** [comment]
> Please add
> 
> `Fixes: 242585656c6bc282f5adbd073e83bafa86b5c0d2`
> 
> to the relevant commit. That's used by backporters to find regression fixes.

Done.

**Comment by vshankar** [comment]
> Please add
> 
> `Fixes: 242585656c6bc282f5adbd073e83bafa86b5c0d2`
> 
> to the relevant commit. That's used by backporters to find regression fixes.

Shouldn't that be `Introduced-by: <commit>`?

**Comment by vshankar** [comment]
jenkins test make check

**Comment by lxbsz** [comment]
> > Please add
> > `Fixes: 242585656c6bc282f5adbd073e83bafa86b5c0d2`
> > to the relevant commit. That's used by backporters to find regression fixes.
> 
> Shouldn't that be `Introduced-by: <commit>`?

Checked the commit history, they are doing like this:

```
commit abd19aa215eeef6da66c522c9fbeec870acee85a
Author: Patrick Donnelly <pdonnell@redhat.com>
Date:   Fri Oct 15 11:08:28 2021 -0400

    qa: reduce frag split confs for dir_split counter test
    
    Fixes: https://tracker.ceph.com/issues/52949
    Fixes: a5675535ba532cfe782238d995751f9bc91f5078
    Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

```

**Comment by vshankar** [comment]
> Checked the commit history, they are all doing like this:

There's also::

```
commit 8f24f42d46d6b32838e8f6ac5c92eb5a32a99824
Author: Ramana Raja <rraja@redhat.com>
Date:   Mon Sep 30 19:21:29 2019 +0530

    mgr/volumes: fix incorrect snapshot path creation
    
    When constructing the snapshot path, the components are a mix of
    byte and string type objects. The 'os.path.join()' method needs
    the components to be of the same type. Hence convert all components
    to byte type objects.
    
    Introduced-by: 3d63cd947f55a8508f972ee249ea6a04836429ae
    Fixes: https://tracker.ceph.com/issues/42096
    Signed-off-by: Ramana Raja <rraja@redhat.com>
```

**Comment by lxbsz** [comment]
> > Checked the commit history, they are all doing like this:
> 
> There's also::
> 
> ```
> commit 8f24f42d46d6b32838e8f6ac5c92eb5a32a99824
> Author: Ramana Raja <rraja@redhat.com>
> Date:   Mon Sep 30 19:21:29 2019 +0530
> 
>     mgr/volumes: fix incorrect snapshot path creation
>     
>     When constructing the snapshot path, the components are a mix of
>     byte and string type objects. The 'os.path.join()' method needs
>     the components to be of the same type. Hence convert all components
>     to byte type objects.
>     
>     Introduced-by: 3d63cd947f55a8508f972ee249ea6a04836429ae
>     Fixes: https://tracker.ceph.com/issues/42096
>     Signed-off-by: Ramana Raja <rraja@redhat.com>
> ```

Hmm, I have updated it by switching to `Introduced-by:`.

**Review by rishabh-d-dave — APPROVED** [review]
I like previous style of multiple parameters on same line more due to it's compactness, but I understand it's not possible in this case.

**Comment by lxbsz** [comment]
> I like previous style of multiple parameters on same line more due to it's compactness, but I understand it's not possible in this case.

Yeah, it will exceed 95 chars for each line, I am afraid the tox check will fail.

**Review by vshankar — APPROVED** [review]

File: qa/tasks/cephfs/test_client_recovery.py [hunk 1]
```diff
@@ -499,10 +499,16 @@ def test_mount_after_evicted_client(self):
 
         # create a new instance of mount_a's class with most of the
         # same settings, but mounted on mount_b's mountpoint.
-        m = cl(self.mount_a.ctx, self.mount_a.test_dir, self.mount_a.client_id,
-               self.mount_a.client_remote, self.mount_a.client_keyring_path,
-               self.mount_b.hostfs_mntpt, self.mount_a.cephfs_name,
-               self.mount_a.cephfs_mntpt, self.mount_a.ceph_brx_net)
+        m = cl(ctx=self.mount_a.ctx,
+               client_config=self.mount_a.client_config,
```
Comment (vshankar) [inline qa/tasks/cephfs/test_client_recovery.py:503]: This fails with kclient runs.

`kernel_mount.py::KernelMount` has the following ctor:

```
    def __init__(self, ctx, test_dir, client_id, client_remote,
                 client_keyring_path=None, hostfs_mntpt=None,
                 cephfs_name=None, cephfs_mntpt=None, brxnet=None, config={}):
```

The parameter is `config`. Moreover, the `config` param is not saved in the instance object, which means using `self.mount_a.client_config` is incorrect too.

**Review by vshankar — CHANGES_REQUESTED** [review]

File: qa/tasks/cephfs/test_client_recovery.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/cephfs/test_client_recovery.py:503]: Fixed it.

**Review by vshankar — APPROVED** [review]

**Comment by lxbsz** [comment]
Rebased it to revolve the conflicts.

**Comment by vshankar** [comment]
jenkins test make check arm64

**Comment by vshankar** [comment]
Failures are - https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-Nov-30

**Pull request merged**